src/components/Popover/PopoverContent.tsx

```diff
@@ -0,0 +1,155 @@
+    /**
+     * Popover content (can be any valid React Element or component)
+     */
+    children: React.ReactNode;
+    /**
+     * Popover Trigger Reference
+     */
+    triggerRef: HTMLElement | HTMLDivElement | null;
+    /**
+     * Is Popover currently displayed?
+     */
+    isShown: boolean;
+    /**
+     * Optional: Specify the Popover content placement (placement changes automatically, if Popover content cannot fit inside viewport with selected placement)
+     */
+    placement?: Placement;
+    /**
+     * Optional: Specify the Popover content offset (margin between Popover trigger and content)
+     */
+    offset?: number;
+    /**
+     * Passes the onClose callback
+     */
+    onClose: (event?: React.KeyboardEvent<HTMLDivElement>) => void;
+}
+
+interface StyledPopoverProps extends React.CSSProperties {
+    isShown: boolean;
+    transform?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    translate?: any;
+    top?: string | number | undefined;
+    left?: string | number | undefined;
+    bottom?: string | number | undefined;
+    right?: string | number | undefined;
+    onKeyDown: (event: React.KeyboardEvent<HTMLDivElement>) => void;
+    ref: MutableRefObject<HTMLDivElement>;
+    tabIndex: number;
+    role: string;
+}
+
+const StyledPopoverParent = styled.div.attrs({ theme })<StyledPopoverProps>`
+    ${({ isShown, transform, top, left, bottom, right, position }) => css`
+        box-sizing: border-box;
+        position: ${position};
+        top: ${top};
+        left: ${left};
+        bottom: ${bottom};
+        right: ${right};
+        transform: ${transform};
+        transition: ${transition(['opacity'], '0.15s', 'ease-in-out')};
+        width: auto;
+        height: auto;
+        z-index: 1000;
+        box-shadow: ${get('shadows.small')};
+        max-height: none;
+        opacity: ${isShown ? '1' : '0'};
+
+        &:focus {
+            outline: 0;
+        }
+    `}
+`;
+
+const PopoverContentContainer = styled(Card)`
+    display: block;
+    padding: ${Spaces[2]};
+`;
+
+export const PopoverContent = ({
+    children,
+    isShown,
+    onClose,
+    offset,
+    placement,
+    triggerRef
+}: PopoverContentProps): JSX.Element => {
+    const popoverRef = React.useRef<HTMLDivElement | null>(null);
+
+    const { styles, update } = usePopper(triggerRef, popoverRef.current, {
+        placement,
+        strategy: 'fixed',
+        modifiers: [
+            {
+                name: 'offset',
+                enabled: !!offset,
+                options: {
+                    offset: [0, offset]
+                }
+            },
+            {
+                name: 'flip',
+                enabled: true
+            }
+        ]
+    });
+
+    const { popper } = styles;
+
+    React.useEffect(() => {
+        // On each update we need to set focus on the popover content, so that we can close the Popover content with ESC key
+        const timer = setTimeout(() => {
```

Comment:
May I ask why the setTimeout?

Reply:
Because we are using useStateWithTimeout to render the Popover content with a little delay (for the transition animation), therefore we need to also wait that little delay before trying to set focus :)

Reply:
Is the animation necessary? I see this logic making the code more prone to bugs and also adding complexity. I'm not sure we generally have animations, so I'm not sure I'd add timeouts to add a barely perceived animation.